### ISSUE TITLE
Can't extend Orm\Query class.

### ISSUE THREAD

**Issue by takyam-git** [fuel/orm#217]
I want to extend and override Orm\Query::select() method.

I tried like this.

``` php
<?php
// create fuel/app/classes/query.php file. and put it.
class Query extends \Orm\Query{
    public function select($add_pks = true){
        $result = call_user_func_array(array($this, 'parent::select'), func_get_args());
        if(is_object($result) and ($result instanceof \Query)){
            //do something
        }
        return $result;
    }
}
```

and put it app/bootstrap.php

``` php
<?php
// fuel/app/bootstrap.php
Autoloader::add_classes(array(
    'Query' => APPPATH.'classes/query.php',
));
```

and I called Model from controller.

``` php
<?php
Model_Foo::find('all');
```

But didn't called my extended Query class.

I researched. 
The reason is \Orm\Model call "Query::forge" insteadof "\Query::forge".

"Query::forge" returns "\Orm\Query" Object.
"\Query::forge" returns "\Query" Object.

Is this design?, or can fix this?

**Comment by WanWizard** [comment]
You can not extend classes the "FuelPHP way" if they are not aliased to global. Which is the case for the ORM package. This is by design.

What kind of functionality do you want to add? It is not very logical to add "business" logic to the ORM query class...

**Comment by takyam-git** [comment]
Thank you. I understood that is a design.

I want to modify query(select) before generate sql. 

this is my want.
http://fuelphp.com/forums/discussion/11885/auto-aes_encrypt-and-aes_decrypt-solution

I will research another approach.

**Comment by WanWizard** [comment]
I'll reopen this as a feature request, it's looks like an interesting feature to support.

Just trying to figure out how to make this platform independent.

**Comment by takyam-git** [comment]
@WanWizard 
Oh!  thanks!!
I hope your work will succeed.

**Comment by WanWizard** [comment]
I've got a workaround for you. After you have loaded the ORM package, you can do

``` php
Autoloader::add_classes(array(
    'Orm\\Query'        => __DIR__.'/classes/myormquery.php',
));
```

It does require you to copy the entire class and then change what's needed, because this will "inject" your class instead of using the original one. You can not extend the original, that would cause a loading loop.

**Comment by takyam-git** [comment]
@WanWizard thank you for response.

I copied `packages/orm/classes/query.php` to `app/classes/orm/query.php` , and add it on last line of `app/bootstrap.php`. 

``` php
Autoloader::add_classes(array(
    'Orm\\Query' => APPPATH.'classes/orm/query.php',
));
```

And I confirmed load the `app/classes/orm/query.php`.This approach isn't best ( when update ), but it met the requirement.

My realy want is the encrypt and decrypt mysql database fields( http://fuelphp.com/forums/discussion/11885/auto-aes_encrypt-and-aes_decrypt-solution ). It was realized by observer. The observer solution is not good too, because execute more queries.

Thank you for let me know!!
